File: python/剑指offer/面试题63. 股票的最大利润.py
from typing import List
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        import sys
        minNum = sys.maxsize
        maxNum = 0
        for i in prices:
            minNum = min(i,minNum)
            maxNum = max(maxNum,i-minNum)
        return maxNum
s = Solution()
prices = [7,1,5,3,6,4]
print(s.maxProfit(prices))

File: go/剑指offer/面试题63. 股票的最大利润.go
package main

func maxProfit(prices []int) int {
	if len(prices) == 0 {
		return 0
	}
	cost := prices[0]
	max := 0
	for i := 1; i < len(prices); i++ {
		if prices[i] < cost {
			cost = prices[i]
		}
		if prices[i]-cost > max {
			max = prices[i] - cost
		}
	}
	return max
}
